yarn-project/simulator/src/private/execution_data_provider.ts

```diff
@@ -54,7 +54,7 @@ export interface ExecutionDataProvider extends CommitmentsDB {
    * @param messageHash - The message hash.
    * @returns A Promise that resolves to an array of field elements representing the auth witness.
    */
-  getAuthWitness(messageHash: Fr): Promise<Fr[]>;
+  getAuthWitness(messageHash: Fr): Promise<Fr[] | undefined>;
```

Comment:
Why this change?

Reply:
The way the data is retrieved now leaves exception management a layer up, so the underlying ones have to deal with undefined values. This is good bc it unifies exception messages between TXE and PXE.